fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/DimensionListPartitionTableDefinition.java

```diff
@@ -0,0 +1,95 @@
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.table;
+
+import com.yahoo.bard.webservice.data.config.ResourceDictionaries;
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.config.names.FieldName;
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.PartitionCompositeTable;
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.resolver.DataSourceFilter;
+import com.yahoo.bard.webservice.table.resolver.DimensionIdFilter;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Holds the fields needed to define a partition physical table.
+ */
+public class DimensionListPartitionTableDefinition extends PhysicalTableDefinition {
+
+    private final Map<TableName, Map<String, Set<String>>> tablePartDefinitions;
+
+    /**
+     * Constructor.
+     *
+     * @param name  Table name of the physical table
+     * @param timeGrain  Zoned time grain of the table
+     * @param metricNames  The Set of metric names on the table
+     * @param dimensionConfigs  Set of dimensions on the table as dimension configs
+     * @param tablePartDefinitions A map from table names to a map of dimension names to sets of values for those
+     * dimensions.  The named table will match if for every dimension named at least one of the set of values is part
+     * of the query.
+     */
+    public DimensionListPartitionTableDefinition(
+            TableName name,
+            ZonedTimeGrain timeGrain,
+            Set<FieldName> metricNames,
+            Set<? extends DimensionConfig> dimensionConfigs,
+            Map<TableName, Map<String, Set<String>>> tablePartDefinitions
+    ) {
+        super(name, timeGrain, metricNames, dimensionConfigs);
+        this.tablePartDefinitions = tablePartDefinitions;
+    }
+
+    @Override
+    public Set<TableName> getDependentTableNames() {
+        return tablePartDefinitions.keySet();
+    }
+
+    @Override
+    public PhysicalTable build(ResourceDictionaries dictionaries, DataSourceMetadataService metadataService) {
+        Map<PhysicalTable, DataSourceFilter> availabilityFilters = tablePartDefinitions.entrySet().stream()
+                .collect(Collectors.toMap(
+                        entry -> dictionaries.getPhysicalDictionary().get(entry.getKey().asName()),
+                        entry -> new DimensionIdFilter(toDimensionValuesMap(
+                                entry.getValue(),
+                                dictionaries.getDimensionDictionary()
```

Comment:
Will it be better to assign `dictionaries.getDimensionDictionary()` and `dictionaries.getPhysicalDictionary()` to variables?